Add scary comment in cpp_custom_type_hack.h

jamesr66a opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#48737 Add scary comment in cpp_custom_type_hack.h**

Differential Revision: [D25280542](https://our.internmc.facebook.com/intern/diff/D25280542)

Commit:
Add scary comment in cpp_custom_type_hack.h

[ghstack-poisoned]

dzhulgakov reviewed (approved):
Lol, sure. Can you make also mark the function "deprecated" in C++ I guess

Commit:
Update on "Add scary comment in cpp_custom_type_hack.h"


Differential Revision: [D25280542](https://our.internmc.facebook.com/intern/diff/D25280542)

[ghstack-poisoned]
